projects/storefrontlib/shared/components/card/card.component.ts

```diff
@@ -101,6 +129,37 @@ export class CardComponent implements OnInit {
     this.editCard.emit(4);
   }
 
+  save(): void {
+    this.saveMode = false;
+    const saveCardEvent: SaveCardEvent = {
+      saveMode: this.saveMode,
+    };
+
+    Object.keys(this.saveForm.controls).forEach((control) => {
+      if (this.saveForm.get(control)?.touched) {
+        const value = this.saveForm.get(control)?.value;
+        if (value) {
+          saveCardEvent[control] = value;
+        }
+        this.saveForm.get(control)?.markAsUntouched();
+        this.saveForm.get(control)?.markAsPristine();
```

Comment:
```suggestion
this.saveForm.get(control)?.reset(value)
```
not 100% sure about this, but maybe you can give it a try.
or even better call the rest on the form group just once, it should reset all controls

https://stackoverflow.com/questions/51449859/how-can-i-reset-a-reactive-form-control-to-originate-state